joysticks.py

```diff
@@ -1,29 +1,38 @@
 """Joystick-related utilities"""
 
-import pygame
 
 class Joysticks:
     """Provides data about usb joysticks (and other
      controllers pygame recognizes as joysticks) via pygame
     """
-    def __init__(self):
-        pygame.init()
 
-        self._joysticks = [pygame.joystick.Joystick(x) for x in range(pygame.joystick.get_count())]
+    def __init__(self, pygame):
+        """Initialize joysticks"""
+
+        self._pygame = pygame
+
+        self._pygame.init()
+
+        self._scan_joysticks()
+
+    def _scan_joysticks(self):
+        # List of all joysticks plugged in (at time of initialization)
+        self._joysticks = [self._pygame.joystick.Joystick(
+            x) for x in range(self._pygame.joystick.get_count())]
         list(map(lambda joy: joy.init(), self._joysticks))
```

Comment:
Why are we wrapping the `map` call with `list` when we aren't making a list of the joysticks? 

Mixing maps and comprehensions? I suppose we could use comprehensions for this too:

    [joy.init() for joy in self._joysticks]

Reply:
`map` just yields values - a little bit like lazy evaluation in Haskell. The `list()` was to force it to actually compute each value. I like using comprehensions all the way through much better, good suggestion.